control-plane-operator/controllers/hostedcontrolplane/clusterpolicy/deployment_test.go

```diff
@@ -0,0 +1,48 @@
+package clusterpolicy
+
+import (
+	hyperv1 "github.com/openshift/hypershift/api/v1beta1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/util"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"testing"
+)
+
+func TestReconcileClusterPolicyDeployment(t *testing.T) {
+
+	imageName := "clusterPolicyImage"
+	// Setup expected values that are universal
+
+	// Setup hypershift hosted control plane.
+	targetNamespace := "test"
+	clusterPolicyDeployment := manifests.ClusterPolicyControllerDeployment(targetNamespace)
+	hcp := &hyperv1.HostedControlPlane{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "hcp",
+			Namespace: targetNamespace,
+		},
+	}
+	hcp.Name = "name"
+	hcp.Namespace = "namespace"
+	ownerRef := config.OwnerRefFrom(hcp)
+
+	testCases := []struct {
+		deploymentConfig config.DeploymentConfig
+	}{
+		// empty deployment config
+		{
+			deploymentConfig: config.DeploymentConfig{},
+		},
+	}
+	for _, tc := range testCases {
+		var expectedTermGraceSeconds *int64 = nil
```

Comment:
how is this probing that `ReconcileDeployment` is not enforcing the values for nil and 0?

Reply:
Maybe we potentially can check the output of `manifest.ClusterPolicyControllerDeployment()` and the same for others and see if those fields are not set there as well? If there are any additional suggestions for that?

Reply:
Can't we set the values explicitly and validate they remain? e.g. pseudocoding:
```
deployment := manifests.ClusterPolicyControllerDeployment(hcp.Namespace)
deployment.Spec.Template.Spec.TerminationGracePeriodSeconds = A
deployment.Spec.MinReadySeconds = B
err := ReconcileDeployment(clusterPolicyDeployment, ownerRef, imageName, tc.deploymentConfig, util.AvailabilityProberImageName, pointer.Int32(1234))
		assert.Equal(t, A, clusterPolicyDeployment.Spec.Template.Spec.TerminationGracePeriodSeconds)
		assert.Equal(t, B, clusterPolicyDeployment.Spec.MinReadySeconds)
```

Reply:
Isn't this literally what you are trying to prove with the PR or did I get it wrong? `This is to ensure that when configurable fields such as minReadySeconds are updated for different deployment strategies, they will not be reset by the hypershift operator in the future`

Reply:
Some of the `ReconcileDeployment` functions sets those fields to nil, but I believe if a field in the `Deployment` struct is nil, then it would leave that field alone, so it should be expected those fields remain nil. Specifically, Openshift API Server's `ReconcileDeployment` function creates a new PodSpec in the function, and so any fields left out would be nil:

https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go#L106

Reply:
I will make the suggested changes to `minReadySeconds` as it looks like 0 is an accepted value and not really an empty one.

Reply:
clusterpolicy.ReconcileDeployment takes what is stored in etcd from the API server, manipulates it, and store it back. If you want to validate ReconcileDeployment doesn't stomp of some particular fields when they are already set please do as in https://github.com/openshift/hypershift/pull/1939/files#r1048795333